pkg/server/httpAPI.go

```diff
@@ -182,3 +189,19 @@ func (h *HTTPHandler) serveInitializationStatus(rw http.ResponseWriter, req *htt
 		h.initializationStatus = initializationStatusNew
 	}
 }
+
+// serveFullSnapshotTrigger triggers an out-of-schedule full snapshot
+// for the configured Snapshotter
+func (h *HTTPHandler) serveFullSnapshotTrigger(rw http.ResponseWriter, req *http.Request) {
+	if h.Snapshotter == nil {
+		h.Logger.Warnf("Ignoring out-of-schedule full snapshot request as snapshotter is not configured")
+		rw.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+	if err := h.Snapshotter.TriggerFullSnapshot(); err != nil {
+		h.Logger.Warnf("Skipped triggering out-of-schedule full snapshot: %v", err)
+		rw.WriteHeader(http.StatusMethodNotAllowed)
```

Comment:
Shouldn't this be StatusInternalServerError  instead as there is a snapshotted configured so the method is allowed but something went wrong while taking the Snapshot.

Reply:
Nice catch @ashwani2k . Thanks!